File: windows/AudioTranscriptionSummary/Models/AuthMethod.cs
// AuthMethod.cs
// AWS 認証方式を表す列挙型

namespace AudioTranscriptionSummary.Models;

/// <summary>
/// AWS 認証方式を表す列挙型
/// - AccessKey: Access Key ID / Secret Access Key による手動入力方式
/// - AwsProfile: AWS CLI プロファイル選択方式（SSO / AssumeRole 対応）
/// </summary>
public enum AuthMethod
{
    /// <summary>Access Key ID / Secret Access Key による手動入力方式</summary>
    AccessKey,

    /// <summary>AWS CLI プロファイル選択方式</summary>
    AwsProfile
}

File: macos/Sources/AudioTranscriptionSummary/Models/AuthMethod.swift
// AuthMethod.swift
// AWS 認証方式を表す列挙型

import Foundation

// MARK: - AuthMethod（認証方式）

/// AWS 認証方式を表す列挙型
/// - `accessKey`: Access Key ID / Secret Access Key による手動入力方式
/// - `awsProfile`: AWS CLI プロファイル選択方式（SSO / AssumeRole 対応）
/// - `sso`: IAM Identity Center（SSO）方式
enum AuthMethod: String, Codable, CaseIterable, Identifiable {
    case sso = "sso"
    case awsProfile = "awsProfile"
    case accessKey = "accessKey"

    /// Identifiable 準拠用の ID
    var id: String { rawValue }

    /// UI 表示用の名前
    var displayName: String {
        switch self {
        case .accessKey:
            return "Access Key"
        case .awsProfile:
            return "AWS Profile"
        case .sso:
            return "IAM Identity Center"
        }
    }
}
